web-ui/src/pages/PulseReportPage.jsx

```diff
@@ -121,34 +136,74 @@ const PulseReportPage = () => {
   }, [dateFrom, dateTo]);
   */
 
+  const average = (arr) => arr.reduce((a, b) => a + b, 0) / arr.length;
+
   // This creates data in the format that recharts needs from pulse data.
   useEffect(() => {
```

Comment:
Lots of data munging in this `useEffect` to prepare data for the line chart, the bar chart, and the five highest and lowest average scores.